### PR TITLE
Simplify some common cases of overflowing integer-multiplication

### PR DESCRIPTION
**Pull request by ptomsich**
While working on the overflowing integer-multiplication for AArch64, I can across the fact that some easy simplifications of integer multiplication hadn't been implemented.

The basic idea is similar to what can be done on a non-overflowing integer-multiply, but I decided to stay away from commutative transforms as these might affect precision when overflowing into floating-point.

### CODE REVIEW

File: hphp/runtime/vm/jit/simplify.cpp [hunk 1]
```diff
@@ -742,6 +742,27 @@ SSATmp* simplifyMulDbl(State& env, const IRInstruction* inst) {
...
+  if (c == 0)
+    return cns(env, 0);
+
+  // X * 1 --> X
+  if (c == 1)
+    return s;
+
+  // X * (-1) --> (0 - X)
+  if (c == -1)
+    return gen(env, SubInt, cns(env, 0), s);
```
Comment (nikic) [inline hphp/runtime/vm/jit/simplify.cpp]: Shouldn't this be `SubIntO` for the X == INT_MIN case?

**Comment by ptomsich** [comment]
Yes, it should.

File: hphp/runtime/vm/jit/simplify.cpp (hunk 1 above)
Comment (ptomsich) [inline hphp/runtime/vm/jit/simplify.cpp]: Turns out that SubIntO on AArch64 doesn't handle that right either... so I'll need to add additional test cases and fix the SubIntO while doing this one.

**Comment by ptomsich** [comment]
No point in keeping this open at this time—I'll revise and reopen a pull request, once done.

**Pull request closed without merging**